### PR TITLE
Updated hall service and updated hall controller

### LINKED ISSUE #7430: Hall of Heroes page is coming up blank
If you click on the link to the left of the Tavern chat, the Hall of Heroes comes up empty.

### PR DESCRIPTION
**Pull request by TheHollidayInn**
Fixes #7430
### Changes

Add hall service with new api-v3
Update hall controller to use the new service

---

UUID:

### CODE REVIEW

**Comment by crookedneighbor** [comment]
What does the currentHeroIndex do? I see it getting set to things, but I don't see it being used.

File: website/client/js/controllers/hallCtrl.js [hunk 1]
```diff
@@ -1,47 +1,53 @@
...
-        $scope.hero = undefined;
-        $scope._heroID = undefined;
-        Hero.query({}, function (heroesData) {
-          $scope.heroes = heroesData.data;
+      Hall.updateHero($scope.hero)
+        .then(function (response) {
+          Notification.text("User updated");
+          $scope.hero = undefined;
+          $scope._heroID = undefined;
+          $scope.heroes[$scope.currentHeroIndex] = response.data.data;
```
Comment (TheHollidayInn) [inline website/client/js/controllers/hallCtrl.js:29]: @crookedneighbor It allows me to update the user without reloading all the heroes (which is what happened before)

Comment (crookedneighbor) [inline website/client/js/controllers/hallCtrl.js:29]: Ahh, nice! Just missed this line looking at it on my phone.

**Comment by crookedneighbor** [comment]
Good work!

**Pull request merged**